clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -285,10 +285,8 @@ public synchronized void assignFromSubscribed(Collection<TopicPartition> assignm
         this.assignment.set(assignedPartitionStates);
     }
 
-    private void registerRebalanceListener(ConsumerRebalanceListener listener) {
-        if (listener == null)
-            throw new IllegalArgumentException("RebalanceListener cannot be null");
-        this.rebalanceListener = listener;
+    private void registerRebalanceListener(Optional<ConsumerRebalanceListener> listener) {
+        this.rebalanceListener = Objects.requireNonNull(listener, "RebalanceListener cannot be null");
```

Comment:
Does Objects.requireNonNull checks the lister.get() or the listener? Since you are passing in an Optional.

Reply:
This is me being very paranoid about the `Optional` itself being `null`. I know, it is a bit confusing 😞